app/controllers/batches_controller.rb

```diff
@@ -18,6 +18,12 @@ def index
     @batches = perform_pagination(@batches)
   end
 
+  def show
+    batch = Batch.find(params[:id])
+    @batch_form = BatchForm.for(batch)
+    return unless authorize_resource(batch, READ_BATCH)
```

Comment:
There must be something missing here, because whatever the authorize result, Rails will still render the `show` template? unless `#authorize_resource` raises? But then the `return` becomes superfluous.